scalasql/core/src/DbApi.scala

```diff
@@ -187,9 +228,16 @@ object DbApi {
       connection: java.sql.Connection,
       config: Config,
       dialect: DialectConfig,
-      autoCommit: Boolean,
-      rollBack0: () => Unit
+      autoCommit: Boolean
   ) extends DbApi.Txn {
+    val listeners = collection.mutable.ArrayDeque.empty[TransactionListener]
```

Comment:
I've considered making `listeners` thread-safe, but the `safepoints` member isn't so it seems more consistent this way.   Client code that requires thread-safety bears the responsibility of synchronizing around mutations to listeners (and safepoints, as before)